python/pyarrow/feather.py

```diff
@@ -42,7 +42,7 @@ def __init__(self, source):
         self.source = source
         self.open(source)
 
-    def read(self, columns=None, nthreads=1):
+    def read(self, columns=None, nthreads=1, as_pandas=True):
```

Comment:
I would prefer if we would deprecate this method completly and move its content to `read_table`. That will make it much more obvious what each method returns. We should keep `read` around for at least a release.